src/e2e-test/test-rule-functions.spec.js

```diff
@@ -0,0 +1,268 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+var assert = require('assert');
+var testRules = require('./test-rule-functions');
+var fs = require('fs');
+var path  = require('path');
+
+describe('testRule', function () {
+
+    it('Rule fhirR4Validation should return empty string when the bundle is a standard FHIR R4 data', function () {
+        var resJson = JSON.parse(fs.readFileSync(path.join(__dirname, './test-samples/FHIR R4/sample1.json')));
+        assert.strictEqual(testRules.fhirR4Validation(resJson).valid, true);
+        assert.strictEqual(testRules.fhirR4Validation(resJson).errorMessage, '');
+    });
+
+    it('Rule fhirR4Validation should return error message when the bundle is not a standard FHIR R4 data', function () {
+        var resJson = {
+            "resourceType": "Bundle",
+            "type": "batch",
+            "entry": [
+                {
+                    "fullUrl": "urn:uuid:40386838-40ff-3f80-b68b-4904de7e7b7b",
+                    "resource": {
+                        "resourceType": "Composition",
+                        "id": "40386838-40ff-3f80-b68b-4904de7e7b7b",
+                        "identifier": {
+                            "use": "official",
+                            "value": "2.16.840.1.113883.19.5.99999.1"
+                        }
+                    }
+                },
+                {
+                    "fullUrl": "urn:uuid:2745d583-e3d1-3f88-8b21-7b59adb60779",
+                    "resource": {
+                        "resourceType": "Patient",
+                        "meta": {
+                            "profile": [
+                                "http://hl7.org/fhir/us/core/StructureDefinition/us-core-patient"
+                            ]
+                        },
+                        "id": "2745d583-e3d1-3f88-8b21-7b59adb60779",
+                    }
+                }
+            ]
+        };
+        assert.strictEqual(testRules.fhirR4Validation(resJson).valid, false);
+        assert.strictEqual(JSON.parse(testRules.fhirR4Validation(resJson).errorMessage).valid, false);
+    });
+
+    it('Rule onePatient should return empty string when there is one Patient resourse', function () {
+        var resJson = {
```

Comment:
Since this bundle is used for several tests could it be made a global variable to save repetition?

Reply:
Right, thanks for your reminder. I have modified it.